libs/accounts/src/lib/accounts-data-provider.ts

```diff
@@ -10,54 +9,73 @@ import type {
   AccountEventsSubscription,
 } from './__generated___/Accounts';
 import { makeDataProvider } from '@vegaprotocol/react-helpers';
+import { AccountType } from '@vegaprotocol/types';
 
-function isAccount(
-  account:
-    | AccountFieldsFragment
-    | IterableElement<AccountEventsSubscription['accounts']>
-): account is AccountFieldsFragment {
-  return (account as AccountFieldsFragment).__typename === 'Account';
+export interface AccountFields extends AccountFieldsFragment {
+  available: string;
+  used: string;
+  deposited: string;
+  balance: string;
+  breakdown?: AccountFields[];
 }
 
-export const getId = (
-  account:
-    | AccountFieldsFragment
-    | IterableElement<AccountEventsSubscription['accounts']>
-) =>
-  isAccount(account)
-    ? `${account.type}-${account.asset.id}-${account.market?.id ?? 'null'}`
-    : `${account.type}-${account.assetId}-${account.marketId}`;
+const USE_ACCOUNT_TYPES = [
+  AccountType.ACCOUNT_TYPE_MARGIN,
+  AccountType.ACCOUNT_TYPE_BOND,
+  AccountType.ACCOUNT_TYPE_FEES_INFRASTRUCTURE,
+  AccountType.ACCOUNT_TYPE_FEES_LIQUIDITY,
+  AccountType.ACCOUNT_TYPE_FEES_MAKER,
+  AccountType.ACCOUNT_TYPE_PENDING_TRANSFERS,
+];
+
+const getData = (responseData: AccountsQuery): AccountFields[] | null => {
+  return (
+    (responseData.party?.accounts &&
+      getAccountData(responseData.party.accounts)) ??
+    null
+  );
+};
+
+const getDelta = (
+  subscriptionData: AccountEventsSubscription
+): AccountEventsSubscription['accounts'] => subscriptionData.accounts;
 
 const update = (
-  data: AccountFieldsFragment[],
+  data: AccountFields[],
   deltas: AccountEventsSubscription['accounts']
-) => {
+): AccountFields[] => {
   return produce(data, (draft) => {
     deltas.forEach((delta) => {
-      const id = getId(delta);
-      const index = draft.findIndex((a) => getId(a) === id);
+      const index = draft.findIndex(
+        (account) => account.asset.id === delta.assetId
+      );
       if (index !== -1) {
-        draft[index].balance = delta.balance;
-      } else {
-        // #TODO handle new account
```

Comment:
chore (non blocking): Suggest leaving this else block in

Reply:
updated